app/src/main/java/com/android/voyageur/ui/search/SearchScreen.kt

```diff
@@ -475,50 +464,69 @@ fun UserSearchResultItem(
               fontSize = 14.sp,
               color = MaterialTheme.colorScheme.onSurfaceVariant,
               maxLines = 1,
-              overflow = TextOverflow.Ellipsis,
-              modifier = Modifier.testTag("userUsername_${user.id}"))
+              overflow = TextOverflow.Ellipsis)
         }
 
-        // Only show the button if the user is not the current user
         if (!isCurrentUser) {
-          Button(
-              onClick = {
-                when {
-                  isContactAdded -> userViewModel.removeContact(user.id)
-                  isRequestPending -> {
-                    // Remove the friend request
-                    val requestId = userViewModel.getSentRequestId(user.id)
-                    if (requestId != null) {
-                      userViewModel.deleteFriendRequest(requestId)
+          Row {
+            if (isRequestReceived) {
+              Button(
+                  onClick = {
+                    val friendRequest = receivedFriendRequests.find { it.from == user.id }
+                    if (friendRequest != null) {
+                      userViewModel.acceptFriendRequest(friendRequest)
                     }
+                  },
+                  enabled = isConnected,
+                  colors =
+                      ButtonDefaults.buttonColors(
+                          containerColor = MaterialTheme.colorScheme.tertiary),
+                  shape = RoundedCornerShape(20.dp),
+                  modifier = Modifier.width(100.dp).height(40.dp)) {
+                    Text(
+                        text = "Accept",
```

Comment:
Could you please use the string resources (`R.string`) for "Accept"?

Reply:
Thank you for pointing that out! You're absolutely right—using string resources (R.string) for "Accept" ensures better maintainability and supports localization. I'll update the code to use stringResource(R.string.accept) for consistency. Appreciate your feedback!